src/backend/distributed/planner/distributed_planner.c

```diff
@@ -2296,9 +2296,24 @@ GetRTEListProperties(List *rangeTableList)
 	RangeTblEntry *rangeTableEntry = NULL;
 	foreach_ptr(rangeTableEntry, rangeTableList)
 	{
-		if (!(rangeTableEntry->rtekind == RTE_RELATION &&
-			  rangeTableEntry->relkind == RELKIND_RELATION))
+		if (rangeTableEntry->rtekind != RTE_RELATION)
```

Comment:
it kind of seems like we'll have these bug whenever we check rtekind.

Is there a well-named function that we can come up with?

Reply:
Actually, we can improve `RegularTable` function and use it along the planner.

As we discussed with @onderkalaci, I was thinking of opening an issue for that as 10.0 item.
What do you think @marcocitus ?

Reply:
@onurctirtir actually suggested having a  function for this as well, but we couldn't find a good name now and didn't want to delay the merge for it now as we considered this like a backport.

Onur will consider having a unified name for this purpose. We currently have `RegularTable()` function, which is also not great

Reply:
maybe we can rename it to `RelkindIsTable` and do the following:

```c
// was RegularTable
bool
RelkindIsTable(Oid relationId)
{
	char relationKind = get_rel_relkind(relationId);
        switch(relationKind)
        {
                case RELKIND_RELATION:
                case RELKIND_FOREIGN_TABLE:
                case RELKIND_PARTITIONED_TABLE:
                        return true;
                default:
                        return false;
        }
}

Reply:
I'll be changing this function to also record MAT_VIEWs on #4292. Can we skip changing it now?